llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -461,18 +496,42 @@ bool AIEHazardRecognizer::checkConflict(
       MemoryBanks, TII->getMemoryCycles(SchedClass), DeltaCycles, std::nullopt);
 }
 
-bool AIEHazardRecognizer::checkConflict(
+ConflictTypeBits AIEHazardRecognizer::checkConflict(
     const ResourceScoreboard<FuncUnitWrapper> &Scoreboard,
     const InstrItineraryData *ItinData, unsigned SchedClass, SlotBits SlotSet,
     MemoryBankBits MemoryBanks, SmallVector<int, 2> MemoryAccessCycles,
     int DeltaCycles, std::optional<int> FUDepthLimit) {
   assert(Scoreboard.isValidDelta(DeltaCycles));
+  ConflictTypeBits Conflict = static_cast<uint32_t>(ConflictType::NoConflict);
+
+  if (checkFormatConflict(Scoreboard, DeltaCycles, SlotSet))
```

Comment:
Is it possible for those functions to return directly the mask? In this case we can have:

```
Conflict |= checkFormatConflict(Scoreboard, DeltaCycles, SlotSet);
....
```
Just one idea....